test/integration/conanfile/test_install_method.py

```diff
@@ -0,0 +1,367 @@
+
+    @pytest.fixture
+    def client(self):
+        tc = TestClient(light=True)
+        tc.save({"dep/conanfile.py": conanfile_dep})
+        tc.run("export dep")
+        return tc
+
+    def test_basic_install_method(self, client):
+        client.run("create dep")
+        layout = client.created_layout()
+        assert layout.package().endswith("p")
+        assert f"Package folder {layout.package()}" in client.out
+        assert f"Running install method in {layout.install()}" in client.out
+        assert f"Running package_info method in {layout.install()}" in client.out
+        client.run("install --requires=dep/1.0")
+        assert f"Running package_info method in {layout.install()}" in client.out
+        # Only issue is that the PackageLayout has no idea about the redirected package folder
+        # So we have to know to check for it in tests, but oh well
+        assert "installed.txt" in os.listdir(layout.install())
+        assert "installed.txt" not in os.listdir(layout.package())
+
+    def test_dependency_install_method(self, client):
+        client.save({"app/conanfile.py": textwrap.dedent("""
+                 from conan import ConanFile
+                 class TestConan(ConanFile):
+                     name = "app"
+                     version = "1.0"
+                     requires = "dep/1.0"
+                     def generate(self):
+                         self.output.info("Running generate method")
+                         dep_pkg_folder = self.dependencies["dep"].package_folder
+                         self.output.info(f"Dep package folder: {dep_pkg_folder}")
+                 """)})
+        client.run("create dep")
+        dep_layout = client.created_layout()
+        client.run("create app")
+        assert f"Dep package folder: {dep_layout.package()}" not in client.out
+        assert f"Dep package folder: {dep_layout.install()}" in client.out
+
+    def test_no_non_info_access(self):
+        client = TestClient(light=True)
+        client.save({"conanfile.py": GenConanfile("dep", "1.0")
+                     .with_install("self.output.info('settings.os: ' + self.settings.os)")})
+        client.run("create .", assert_error=True)
+        assert "'self.settings' access in 'install()' method is forbidden" in client.out
+
+    def test_install_moves_from_package(self):
+        client = TestClient(light=True)
+        client.save({"conanfile.py": GenConanfile("dep", "1.0")
+                     .with_import("from conan.tools.files import save, rename",
+                                  "import os")
+                     .with_option("move", [True, False])
+                     .with_package('save(self, os.path.join(self.package_folder, "file.txt"), "Hello World!")',
+                                   "save(self, os.path.join(self.package_folder, 'file2.txt'), 'Hello World 2!')")
+                     # This is NOT allowed, moving from package to install is forbidden, only as test to ensure consistency
```

Comment:
This is something that should not be allowed, by this point the package id has already been compuetd, so modifying its contents breaks cache integrity as shown below